book/src/libs/wasp/benchspy/first_test.md

```diff
@@ -0,0 +1,113 @@
+# BenchSpy - Your First Test
+
+Let's start with the simplest case, which doesn't require any part of the observability stack—only `WASP` and the application you are testing.
+`BenchSpy` comes with built-in `QueryExecutors`, each of which also has predefined metrics that you can use. One of these executors is the `DirectQueryExecutor`, which fetches metrics directly from `WASP` generators,
+which means you can run it with Loki.
+
+> [!NOTE]
+> Not sure whether to use `Loki` or `Direct` query executors? [Read this!](./loki_dillema.md)
+
+## Test Overview
+
+Our first test will follow this logic:
+- Run a simple load test.
+- Generate a performance report and store it.
+- Run the load test again.
+- Generate a new report and compare it to the previous one.
+
+We'll use very simplified assertions for this example and expect the performance to remain unchanged.
+
+### Step 1: Define and Run a Generator
+
+Let's start by defining and running a generator that uses a mocked service:
+
+```go
+gen, err := wasp.NewGenerator(&wasp.Config{
+    T:           t,
+    GenName:     "vu",
+    CallTimeout: 100 * time.Millisecond,
+    LoadType:    wasp.VU,
+    Schedule:    wasp.Plain(10, 15*time.Second),
+    VU: wasp.NewMockVU(&wasp.MockVirtualUserConfig{
+        CallSleep: 50 * time.Millisecond,
+    }),
+})
+require.NoError(t, err)
+gen.Run(true)
+```
+
+### Step 2: Generate a Baseline Performance Report
+
+With load data available, let's generate a baseline performance report and store it in local storage:
+
+```go
+baseLineReport, err := benchspy.NewStandardReport(
+    // random hash, this should be the commit or hash of the Application Under Test (AUT)
+    "e7fc5826a572c09f8b93df3b9f674113372ce924",
+    // use built-in queries for an executor that fetches data directly from the WASP generator
+    benchspy.WithStandardQueries(benchspy.StandardQueryExecutor_Direct),
+    // WASP generators
+    benchspy.WithGenerators(gen),
+)
+require.NoError(t, err, "failed to create baseline report")
+
+fetchCtx, cancelFn := context.WithTimeout(context.Background(), 60*time.Second)
+defer cancelFn()
+
+fetchErr := baseLineReport.FetchData(fetchCtx)
+require.NoError(t, fetchErr, "failed to fetch data for baseline report")
+
+path, storeErr := baseLineReport.Store()
+require.NoError(t, storeErr, "failed to store baseline report", path)
+```
+
+> [!NOTE]
+> There's a lot to unpack here, and you're encouraged to read more about the built-in `QueryExecutors` and the standard metrics they provide as well as about the `StandardReport` [here](./reports/standard_report.md).
+>
+> For now, it's enough to know that the standard metrics provided by `StandardQueryExecutor_Direct` include:
+> - Median latency
+> - P95 latency (95th percentile)
```

Comment:
I'd say let's better add max latency to get extreme outliers. 95th and 99th are usually product requirements but you can rely on one or another and discuss it with stakeholders. MAX on the other hand shows you extreme outliers and if you have a strict SLA that, for example, "0 transactions are delivered after 2 minutes" it will show you that.